contracts/extensions/collections/HasProxyAdmin.sol

```diff
@@ -4,18 +4,32 @@ pragma solidity ^0.8.0;
 import "@openzeppelin/contracts/utils/StorageSlot.sol";
 
 abstract contract HasProxyAdmin {
+  error Unauthorized();
+
   // bytes32(uint256(keccak256("eip1967.proxy.admin")) - 1));
   bytes32 internal constant _ADMIN_SLOT = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
 
   modifier onlyAdmin() {
-    require(msg.sender == _getAdmin(), "HasProxyAdmin: unauthorized sender");
+    _requiresAdmin();
     _;
   }
 
+  function _requiresAdmin() internal view {
+    assembly {
+      if iszero(eq(caller(), sload(_ADMIN_SLOT))) {
+        /// @dev value is equal to bytes4(keccak256("Unauthorized()"))
+        mstore(0x00, 0x82b42900)
+        revert(0x1c, 0x04)
+      }
+    }
+  }
+
   /**
-   * @dev Returns proxy admin.
+   * @dev Returns proxy admin.ƒ
    */
-  function _getAdmin() internal view returns (address) {
-    return StorageSlot.getAddressSlot(_ADMIN_SLOT).value;
+  function _getAdmin() internal view returns (address admin) {
+    assembly {
+      admin := sload(_ADMIN_SLOT)
+    }
```

Comment:
I think we can remove Line 4 `import "@openzeppelin/contracts/utils/StorageSlot.sol";` for better code size saving here, since the library is not used anymore.